src/java.base/share/classes/sun/security/pkcs/PKCS9Attributes.java

```diff
@@ -241,10 +241,8 @@ public void encode(byte tag, OutputStream out) throws IOException {
 
     private byte[] generateDerEncoding() throws IOException {
         DerOutputStream out = new DerOutputStream();
-        Object[] attribVals = attributes.values().toArray();
-
-        out.putOrderedSetOf(DerValue.tag_SetOf,
-                            castToDerEncoder(attribVals));
+        DerEncoder[] attribVals = attributes.values().toArray(new DerEncoder[0]);
```

Comment:
I suggest to use `DerEncoder::new` here and in other places, just make sure that class initialization order will be the same before and after this patch with `java -Xlog:class+load -version`